mesonbuild/coredata.py

```diff
@@ -184,6 +182,46 @@ def __init__(self, options):
         # Only to print a warning if it changes between Meson invocations.
         self.pkgconf_envvar = os.environ.get('PKG_CONFIG_PATH', '')
 
+    @staticmethod
+    def __load_cross_file(filename):
+        """Try to load the cross file.
+
+        If the filename is None return None. If the filename is an absolute
+        (after resolving variables and ~), return that absolute path. Next,
+        check if the file is relative to the current source dir. If the path
+        still isn't resolved do the following:
+            Linux + BSD:
+                - $XDG_DATA_HOME/meson/cross (or ~/.local/share/meson/cross if
+                  undefined)
+                - $XDG_DATA_DIRS/meson/cross (or
+                  /usr/local/share/meson/cross:/usr/share/meson/cross if undefined)
+                - Error
+            *:
+                - Error
+        BSD follows the Linux path and will honor XDG_* if set. This simplifies
+        the implementation somewhat, especially since most BSD users wont set
+        those environment variables.
+        """
+        if filename is None:
+            return None
+        filename = os.path.expanduser(os.path.expandvars(filename))
+        if os.path.isabs(filename):
+            return filename
+        path_to_try = os.path.abspath(filename)
+        if os.path.exists(path_to_try):
+            return path_to_try
+        if sys.platform == 'linux' or 'bsd' in sys.platform.lower():
+            paths = [
+                os.environ.get('XDG_DATA_HOME', os.path.join(os.environ['HOME'], '.local', 'share')),
```

Comment:
you can use `os.expand("~/.local/share")`

Reply:
Did you mean `os.path.expanduser("~/.local/share")` I'll do that if that's preferred.